config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/templates/cluster-api-provider-vsphere-periodics-upgrades.yaml.tpl

```diff
@@ -1,7 +1,17 @@
-{{ if eq $.branch "release-1.5" "release-1.6" "release-1.7" "release-1.8" | not }}
+{{ $modes := list "-govmomi" "-supervisor" -}}
+{{ range $i, $mode := $modes -}}
+{{ $modeFocus := "" -}}
+{{ if eq $mode "-supervisor" }}{{ $modeFocus = "\\\\[supervisor\\\\] " }}{{ end -}}
+{{/* e2e upgrade has been introduced in release-1.9 */ -}}
+{{/* e2e upgrade in supervisor mode has been introduced in release-1.10 */ -}}
+{{ $skipInBranch := list "release-1.5" "release-1.6" "release-1.7" "release-1.8" -}}
+{{ if eq $mode "-supervisor" }}{{ $skipInBranch = list "release-1.5" "release-1.6" "release-1.7" "release-1.8" "release-1.9" }}{{ end -}}
+{{ if has $.branch $skipInBranch | not -}}
+{{ if eq $i 0 -}}
 periodics:
```

Comment:
Let's just move periodics to l.1

Reply:
This is not possible, because it will start generating files also for branches where we are not running any periodic upgrade tests